gsa/src/web/pages/alerts/__tests__/detailspage.js

```diff
@@ -0,0 +1,517 @@
+/* Copyright (C) 2019-2020 Greenbone Networks GmbH
+ *
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation, either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+import React from 'react';
+
+import {setLocale} from 'gmp/locale/lang';
+
+import {isDefined} from 'gmp/utils/identity';
+
+import Capabilities from 'gmp/capabilities/capabilities';
+import CollectionCounts from 'gmp/collection/collectioncounts';
+
+import Alert from 'gmp/models/alert';
+import Filter from 'gmp/models/filter';
+
+import {createRenewSessionQueryMock} from 'web/graphql/__mocks__/session';
+
+import {entityLoadingActions} from 'web/store/entities/alerts';
+import {setTimezone, setUsername} from 'web/store/usersettings/actions';
+
+import {rendererWith, fireEvent, screen, wait} from 'web/utils/testing';
+
+import Detailspage, {ToolBarIcons} from '../detailspage';
+
+setLocale('en');
+
+if (!isDefined(window.URL)) {
+  window.URL = {};
+}
+window.URL.createObjectURL = jest.fn();
```

Comment:
Is this code used for the tests. As far as I can see no download code paths are tested and therefore createObjectURL is not called. But I am not 100 percent sure.

Reply:
In the "should call commands" test the export icon is clicked which uses the download component. Removing the code would cause the test to fail because window.URL.createObjectURL is not defined.